AWSTranscribeStreamingUnitTests/AWSTranscribeStreamingClientWebSocketProviderTests.swift

```diff
@@ -0,0 +1,332 @@
+            }
+        }
+        delegate.receiveEventCallback = { data, error in
+            if error == nil {
+                receiveDataSend.fulfill()
+            }
+        }
+        
+        client.setDelegate(delegate, callbackQueue: DispatchQueue.global())
+        
+        client.startTranscriptionWSS(request)
+        //add waiter to open web socket and make sure its connected before sending data.
+        
+        wait(for: [receiveDataSend], timeout: 10)
+    }
+}
+
+class MockAWSTranscribeStreamingClientDelegate: NSObject, AWSTranscribeStreamingClientDelegate {
+    func connectionStatusDidChange(_ connectionStatus: AWSTranscribeStreamingClientConnectionStatus, withError error: Error?) {
+        
+    }
+    
+    func didReceiveEvent(_ event: AWSTranscribeStreamingTranscriptResultStream?, decodingError: Error?) {
+        
+    }
+}
+
+class MockWebSocketProvider: NSObject, AWSTranscribeStreamingWebSocketProvider{
+    
+    var webSocket: AWSTranscribeStreamingWebSocketProvider
+    
+    var clientDelegate: AWSTranscribeStreamingClientDelegate
+    
+    var delegate: MockWebSocketProviderDelegate?
+    var errorOnConnectionStatusDidChange: NSError?
+    
+    override init() {
+        clientDelegate = MockAWSTranscribeStreamingClientDelegate()
+        webSocket = AWSSRWebSocket()
+        delegate = nil
+        errorOnConnectionStatusDidChange = nil
+        super.init()
+    }
+    
+    func setErrorOnConnectionStatusDidChange(error: NSError?) {
+        self.errorOnConnectionStatusDidChange = error
+    }
+    
+    func send(_ data: Data) {
+        self.delegate?.webSocket(webSocket, didReceiveMessage: data)
+    }
+    
+    func connect() {
+        self.delegate?.webSocketDidOpen(webSocket)
+    }
+    
+    func disconnect() {
+        self.delegate?.webSocketDidCloseWithCode(webSocket, code: 0, reason: "clean", wasClean: true)
+    }
+    
+    func setDelegate(_ delegate: AWSTranscribeStreamingClientDelegate, dispatchQueue: DispatchQueue) {
+        self.clientDelegate = delegate
+        let adaptor = MockWebSocketProviderDelegate(clientDelegate: delegate,
+                                                    callbackQueue: dispatchQueue,
+                                                    errorOnConnectionStatusDidChange: self.errorOnConnectionStatusDidChange)
+        self.delegate = adaptor
+    }
+    
+    func configure(with urlRequest: URLRequest) {
+        self.webSocket = AWSSRWebSocket(urlRequest: urlRequest)
+    }
+}
+
+class MockReturningCredentialsProvider: NSObject, AWSCredentialsProvider {
+    func credentials() -> AWSTask<AWSCredentials> {
+        let credentials = AWSCredentials(accessKey: "memem", secretKey: "hahah", sessionKey: "lskdjflkd", expiration: nil)
+        return AWSTask(result: credentials)
+    }
+    
+    func invalidateCachedTemporaryCredentials() {
+        // Do nothing
+    }
+}
+
+class MockWebSocketProviderDelegate: NSObject {
```

Comment:
I think you can remove this and set your mock responses on the MockProvider?

Reply:
refactored and removed